apps/studio/src/features/editing-experience/components/LayoutSelection.tsx

```diff
@@ -0,0 +1,245 @@
+// Component for layout selection
+
+import React, { useState } from 'react'
+import {
+  Box,
+  Grid,
+  GridItem,
+  HStack,
+  Image,
+  Text,
+  VStack,
+  Icon,
+} from '@chakra-ui/react'
+import { Button, Link } from '@opengovsg/design-system-react'
+import { BiLeftArrowAlt, BiRightArrowAlt, BiShow } from 'react-icons/bi'
+import Preview, {
+  type PreviewProps,
+} from '~/features/editing-experience/components/Preview'
+import articleLayoutPreview from '~/features/editing-experience/data/articleLayoutPreview.json'
+import contentLayoutPreview from '~/features/editing-experience/data/contentLayoutPreview.json'
+
+export interface LayoutSelectionProps {
+  pageName: string
+  // preparing to make the trpc call to create page. required props
+  pageUrl: string
+  siteId: string
+  folderId: string
+}
+
+const LAYOUT_DATA = [
+  {
+    layoutDisplayName: 'Default',
+    layoutTypename: 'content',
+    layoutDescription: 'This is the most basic layout for your content.',
+    imageSource: '/assets/layout-card/default_layout_card.webp',
+    previewJson: contentLayoutPreview,
+  },
+  {
+    layoutDisplayName: 'Article',
+    layoutTypename: 'article',
+    layoutDescription:
+      'Designed for the perfect reading experience. Use this layout for text-heavy content, such as news, press releases, and speeches',
+    imageSource: '/assets/layout-card/article_layout_card.webp',
+    previewJson: articleLayoutPreview,
+  },
+] as const
+
+const dataAttr = (value: unknown) => (!!value ? true : undefined) // Keeping this here for now, might shift out to a utils file when necessary.
+
+// TODO: Make this headless by getting the LAYOUT_DATA from the schema. Find somewhere in the schema for layoutDescription & image(fetch this too or generate it)
+function LayoutSelection(props: LayoutSelectionProps): JSX.Element {
+  const [selectedLayout, setSelectedLayout] = useState<LayoutDataType>(
+    LAYOUT_DATA[0]!,
+  )
+
+  return (
+    <>
+      {/* This is the floating header */}
+      <VStack position="fixed" h="100%">
```

Comment:
nit: instead of using fixed, may be better to use `position="sticky"` (and top=0) instead. Then the grid below will not need to have a `calc(100% - 7rem)` prop. Of course, the parent of this needs to be relative.

Reply:
Tried to use position="sticky" but can't seem to get it to work with the independent overflow scrolling as I had difficulty forcing the outer container height to be 100%, as it tends to expand to fit contents resulting in non independent scrolling.

Reply:
i think since this is a nit and you had difficulty getting it to work, feel free to resolve this and merge this PR in! @hanpuliu-charles